File: Assets/Scripts/Minion Scripts/MinionManager.js
#pragma strict
@script RequireComponent(GenericCreature);
@script RequireComponent(SphereCollider);

var GenCre: GenericCreature;
var BasicAttackObj: GameObject;
var SpCo: SphereCollider;

function Start () {
	SpCo = GetComponent(SphereCollider);
	GenCre = GetComponent(GenericCreature);
	GenCre.DeathFunc = DeathFunction;
	GenCre.objHeight =  SpCo.radius*2;
	//GenCre.BasicAttackObj = BasicAttackObj;  //I have commented this out because I set a default on the prefab (Refrain from doing that, btw) until there is a projectile for minions.
}

function Update () {

}

function DeathFunction(){
	if(Network.isServer){
		Network.Destroy(networkView.viewID);
		//TODO: Add the rest :P
	}
}

File: Assets/Scripts/Minion Scripts/MinionManager.cs
using UnityEngine;
using System.Collections;
[RequireComponent (typeof (SphereCollider))]

public class MinionManager : UnitBase {
	//@script RequireComponent(GenericCreature);
	//@script RequireComponent(SphereCollider);
	SphereCollider SpCo;

	// Use this for initialization
	public override void Start () {
		base.Start ();
		SpCo = (GetComponent(typeof(SphereCollider)) as SphereCollider);

		objHeight = SpCo.radius*2;
		objWidth = SpCo.radius*2;
	}
	
	// Update is called once per frame
	public override void Update () {
		base.Update();
	}

	[RPC]
	public override IEnumerator Die(){
		if(Network.isServer){
			Network.Destroy(networkView.viewID);
			//TODO: Add the rest :P
		}

		yield return new WaitForEndOfFrame();
	}
}
